src/Generator/Handler/GenerateZoneHandler.php

```diff
@@ -4,59 +4,48 @@
 
 namespace App\Generator\Handler;
 
-use App\Entity\Mycelium;
-use App\Entity\MyceliumGenusEnum;
-use App\Entity\Zone;
+use App\Generator\Message\GenerateTreeMessage;
 use App\Generator\Message\GenerateZoneMessage;
-use App\Repository\MyceliumRepository;
-use App\Repository\ZoneRepository;
-use Doctrine\ORM\EntityManagerInterface;
+use App\Repository\TreeRepository;
+use RuntimeException;
 use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+use Symfony\Component\Messenger\Exception\ExceptionInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
 
 #[AsMessageHandler]
 class GenerateZoneHandler
 {
-    private MyceliumRepository $myceliumRepository;
-    private ZoneRepository $zoneRepository;
-    private EntityManagerInterface $entityManager;
+    private TreeRepository $treeRepository;
+    private MessageBusInterface $messageBus;
 
     /**
-     * @param EntityManagerInterface $entityManager
-     * @param MyceliumRepository     $myceliumRepository
-     * @param ZoneRepository         $zoneRepository
+     * @param MessageBusInterface    $messageBus
+     * @param TreeRepository         $treeRepository
      */
-    public function __construct(
-        EntityManagerInterface $entityManager,
-        MyceliumRepository $myceliumRepository,
-        ZoneRepository $zoneRepository
-    ) {
-        $this->entityManager = $entityManager;
-        $this->myceliumRepository = $myceliumRepository;
-        $this->zoneRepository = $zoneRepository;
+    public function __construct(MessageBusInterface $messageBus, TreeRepository $treeRepository)
+    {
+        $this->treeRepository = $treeRepository;
+        $this->messageBus = $messageBus;
     }
 
     /**
      * @param GenerateZoneMessage $generateZoneMessage
      *
      * @return void
+     * @throws ExceptionInterface
      */
     public function __invoke(GenerateZoneMessage $generateZoneMessage): void
```

Comment:
todo:
- split in tree message as generateZoneTrees, generateZoneSporocarps and generateZoneMyceliums